runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -0,0 +1,359 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.worldhopper;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ObjectArrays;
+import com.google.common.eventbus.Subscribe;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.ChatPlayer;
+import net.runelite.api.ClanMember;
+import net.runelite.api.Client;
+import net.runelite.api.Friend;
+import net.runelite.api.GameState;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.api.events.PlayerMenuOptionClicked;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.chat.ChatColorType;
+import net.runelite.client.chat.ChatMessageBuilder;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.chat.QueuedMessage;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.Text;
+import net.runelite.http.api.worlds.World;
+import net.runelite.http.api.worlds.WorldClient;
+import net.runelite.http.api.worlds.WorldResult;
+import net.runelite.http.api.worlds.WorldType;
+import org.apache.commons.lang3.ArrayUtils;
+
+@PluginDescriptor(
+	name = "World Hopper",
+	description = "Allows you to quickly hop worlds"
+)
+@Slf4j
+public class WorldHopperPlugin extends Plugin
+{
+	private static final String HOP_TO = "Hop-to";
+	private static final String KICK_OPTION = "Kick";
+	private static final ImmutableList<String> BEFORE_OPTIONS = ImmutableList.of("Add friend", "Remove friend", KICK_OPTION);
+	private static final ImmutableList<String> AFTER_OPTIONS = ImmutableList.of("Message");
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private KeyManager keyManager;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Inject
+	private ScheduledExecutorService executorService;
+
+	@Inject
+	private WorldHopperInputListener inputListener;
+
+	private ScheduledFuture<?> lookupFuture;
+	private WorldResult worldResult;
+
+	private int worldToHopTo = -1;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(inputListener);
+
+		lookupFuture = executorService.scheduleAtFixedRate(() ->
+		{
+			try
+			{
+				WorldResult worldResult = new WorldClient().lookupWorlds();
```

Comment:
You can just use client API for this, no need to use WorldClient at all

Reply:
I am pretty sure the client API for world list doesn't work unless the world switcher menu is open.. so we can't use that. This seems fine. Additionally we have seen that they cannot change the world list without restarting everything so I probably would not bother scheduling this at all.